src/Core/Checkout/Cart/Order/OrderConverter.php

```diff
@@ -265,34 +266,61 @@ public function assembleSalesChannelContext(OrderEntity $order, Context $context
         }
 
         $customerId = $order->getOrderCustomer()->getCustomerId();
-        $customerGroupId = null;
+        /** @var CustomerEntity|null $customer */
+        $customer = null;
 
         if ($customerId) {
-            $customer = $this->customerRepository->search(new Criteria([$customerId]), $context)->getEntities()->get($customerId);
-            if ($customer !== null) {
-                $customerGroupId = $customer->getGroupId();
-            }
+            $customerCriteria = new Criteria([$customerId]);
+            $customerCriteria->addAssociation('addresses');
+
+            $customer = $this->customerRepository->search($customerCriteria, $context)->first();
```

Comment:
you probably need to do `getEntities()->first()` for the aforementioned to be true.